### PR TITLE
feat(grouping): Add more variables to substitution for fingerprints

### PR DESCRIPTION
**Pull request by mitsuhiko**
Adds the following variable substitutions for server side fingerprinting:

* `stack.abs_path` (or `path`): this one was missing but should have existed
* `stack.filename`: like `stack.abs_path` but inverted (prefers filename)

These two are somewhat dubious in usefulness but one could expect them to exist even if they are footguns:

* `error.value` or (`value`)
* `message`

### CODE REVIEW

File: src/sentry/grouping/utils.py [hunk 1]
```diff
@@ -42,13 +42,31 @@ def get_rule_bool(value):
 def get_fingerprint_value(var, data):
     if var == "transaction":
         return data.get("transaction") or "<no-transaction>"
+    elif var == "message":
+        message = (
+            get_path(data, "logentry", "formatted")
+            or get_path(data, "logentry", "message")
+            or get_path(data, "exception", "values", -1, "value")
+        )
+        return message or "<no-message>"
```
Comment (jan-auer) [inline src/sentry/grouping/utils.py:51]: I believe this should prioritize `logentry.message` and use the cleaned up message where all replacements of dynamic values have run.

Comment (mitsuhiko) [reply src/sentry/grouping/utils.py:51]: The main use of `message` here is to be displayed in the title, not to be used for grouping. This code is kept aligned with how it is used for the title generation.

File: src/sentry/grouping/utils.py [hunk 2]
```diff
@@ -42,13 +42,31 @@ def get_rule_bool(value):
...
+        return value or "<no-value>"
     elif var in ("function", "stack.function"):
         frame = get_crash_frame_from_event_data(data)
         func = frame.get("function") if frame else None
         return func or "<no-function>"
+    elif var in ("path", "stack.abs_path"):
+        frame = get_crash_frame_from_event_data(data)
+        func = frame.get("abs_path") or frame.get("filename") if frame else None
+        return func or "<no-abs-path>"
+    elif var == "stack.filename":
```
Comment (jan-auer) [inline src/sentry/grouping/utils.py:66]: Could alias to `filename`.

**Review by jan-auer — APPROVED** [review]

File: src/sentry/grouping/utils.py (hunk 2 above)
Comment (mitsuhiko) [reply src/sentry/grouping/utils.py:66]: Want to phase out the short versions as much as possible since they are inconsistent with the product elsewhere.

**Pull request merged**